atomic_reactor/util.py

```diff
@@ -867,41 +907,21 @@ def get_manifest_digests(image, registry, insecure=False, dockercfg_path=None,
             # media type
             elif (ex.response.status_code == requests.codes.not_found or
                   ex.response.status_code == requests.codes.not_acceptable):
+                logger.debug("skipping version %s due to status code %s",
+                             version, ex.response.status_code)
                 continue
             else:
                 raise
 
-        received_media_type = None
-        try:
-            received_media_type = response.headers['Content-Type']
-        except KeyError:
-            # Guess content_type from contents
-            try:
-                encoding = guess_json_utf(response.content)
-                manifest = json.loads(response.content.decode(encoding))
-                received_media_type = manifest['mediaType']
-            except (ValueError,  # not valid JSON
-                    KeyError) as ex:  # no mediaType key
-                logger.warning("Unable to fetch media type: neither Content-Type header "
-                               "nor mediaType in output was found")
-
-        if not received_media_type:
-            continue
-
-        # Only compare prefix as response may use +prettyjws suffix
-        # which is the case for signed manifest
-        response_h_prefix = received_media_type.rsplit('+', 1)[0]
-        request_h_prefix = media_type.rsplit('+', 1)[0]
-        if response_h_prefix != request_h_prefix:
-            logger.debug('request headers: %s', headers)
-            logger.debug('response headers: %s', response.headers)
-            logger.warning('Received media type %s mismatches the expected %s',
-                           received_media_type, media_type)
-            continue
+        if version != 'v1':
+            if not manifest_is_media_type(response, media_type):
+                logger.error("content does not match expected media type")
+                continue
+            else:
+                logger.debug("content matches expected media type")
```

Comment:
Nit: assign the return of manifest_is_media_type to a variable and print out the expected and actual values in the error message for simpler debugging. and put the expected media type in the logger.debug message for consistency.

Reply:
We don't have the returned/guess media type by this point, and we want to log at different levels depending on the result (a match is not an error), so I'm not sure how you'd like this to look.

Reply:
Sorry, for some reason I thought manifest_is_media_type returned something other than a Boolean. Ignore the request.